akd_core/src/utils.rs

```diff
@@ -16,6 +16,47 @@ pub fn get_marker_version_log2(version: u64) -> u64 {
     64 - (version.leading_zeros() as u64) - 1
 }
 
+/// Return two (possibly empty) lists of marker versions, given
+/// a start_version and end_version for the range of update proofs, along with
+/// a final epoch.
+///
+/// The first list contains versions which should be checked for membership,
+/// and the second list contains versions which should be checked for non-membership.
+///
+/// Roughly, the intervals should be organized as follows:
+/// 1 --- { previous marker versions } --- [start_version, end_version] --- { future marker versions } --- epoch
+///
+/// In this implementation, the set of previous marker versions consists of the largest power of 2
+/// that is at most start_version (or is empty if start_version is already a power of 2). The set of
+/// future marker versions is as described in SEEMless: the consecutively increasing set of versions
+/// from end_version until the next power of 2, and then all consecutive powers of 2 up until the
+/// epoch.
+///
+/// This will panic if start_version = 0
+pub fn get_marker_versions(
+    start_version: u64,
+    end_version: u64,
+    epoch: u64,
+) -> (Vec<u64>, Vec<u64>) {
```

Comment:
Nit: I don't know if a tuple of `Vec` results feels super intuitive to me. I think I'd prefer named fields here to make it a bit more apparent to readers what these represent.

Reply:
sounds good!